Remove the transport client

rjernst opened the pull request:
This commit removes the transport client and all remaining uses in the code.

rjernst committed:
Remove the transport client

This commit removes the transport client and all remaining uses in the code.

elasticmachine commented:
Pinging @elastic/es-core-infra

rjernst committed:
fixes

jasontedor commented on server/src/main/java/org/elasticsearch/cluster/ClusterState.java:
```diff
@@ -132,8 +131,7 @@
                 final String requiredFeature = custom.getRequiredFeature().get();
-                // if it is a transport client we are lenient yet for a connected node it must have the required feature
-                return out.hasFeature(requiredFeature) || out.hasFeature(TransportClient.TRANSPORT_CLIENT_FEATURE) == false;
+                return out.hasFeature(requiredFeature);
```
There are changes needed for the Javadoc here. It might be worth grepping the codebase for “transport client”.

rjernst replied:
I did another pass and removed lots of uses. I rewrote some comments referring to the transport client as TODOs to consider modifying the code in question, but leave those for followups.

jasontedor reviewed (changes requested):
I left a comment.

jasontedor commented on server/src/test/java/org/elasticsearch/persistent/PersistentTasksCustomMetaDataTests.java:
```diff
@@ -272,12 +271,8 @@ public void testMinVersionSerialization() throws IOException {
-        if (transportClient) {
-            features.add(TransportClient.TRANSPORT_CLIENT_FEATURE);
-        }
         // if a transport client, then it must have the feature otherwise we add the feature randomly
```
This comment is stale now.

rjernst committed:
remove more references to transport client

jasontedor commented on TESTING.asciidoc:
```diff
@@ -305,12 +305,6 @@ e.g. -Dtests.rest.suite=index,get,create/10_with_id
 blacklisted and need to be skipped
 e.g. -Dtests.rest.blacklist=index/*/Index document,get/10_basic/*
 
-Note that the REST tests, like all the integration tests, can be run against an external
```
Dropping this paragraph loses that `tests.cluster` applies to the REST tests too.

rjernst replied:
The rest tests use `tests.rest.cluster`

rjernst committed:
Merge branch 'master' into remove_transport_client4

rjernst committed:
Merge branch 'master' into remove_transport_client4

rjernst committed:
Use project def now for matrix stats

rjernst committed:
Make high level rest client a fat jar

The original intention of the high level rest client was to provide a
single jar. We tried this long ago, but had issues with intellij not
correctly resolving internal tests that relied on the HLRC. This
commit tweaks our use of the shadow plugin so we now produce a correct
fat jar (minus the LLRC and server jars, which we can address later),
with the module "client" dependencies included, as well as the
correct pom file omitting those dependencies.

relates #42638

rjernst committed:
Merge branch 'master' into remove_transport_client4

rjernst committed:
fix tests

mark-vieira committed:
Ensure we don't overwrite original jar with fatjar

mark-vieira committed:
Exclude shadowjar bundled dependencies from test runtime classpath

rjernst committed:
Merge branch 'hlrc_deps' into remove_transport_client4

rjernst committed:
Merge branch 'master' into remove_transport_client4

rjernst committed:
add back server dep

rjernst committed:
fix null env uses in test

rjernst committed:
more test fixes

rjernst commented:
@elasticmachine update branch

elasticmachine committed:
Merge branch 'master' into remove_transport_client4

jaymode commented on x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ssl/TrustConfig.java:
```diff
@@ -38,15 +38,15 @@
-     * @param environment the environment to resolve files against or null in the case of running in a transport client
+     * @param environment the environment to resolve files against
      */
     abstract List<Path> filesToMonitor(@Nullable Environment environment);
```
This nullable can be removed I think

jaymode reviewed (approved):
LGTM

rjernst committed:
Merge branch 'master' into remove_transport_client4

rjernst committed:
remove nullable

rjernst committed:
Merge branch 'master' into remove_transport_client4

jasontedor reviewed (approved):
LGTM.

rjernst committed:
Merge branch 'master' into remove_transport_client4
